stdlib/src/collections/list.mojo

```diff
@@ -790,3 +790,61 @@ struct List[T: CollectionElement](CollectionElement, Sized, Boolable):
             if i[] == rebind[T2](value):
                 return True
         return False
+
+    @always_inline
+    fn __eq__[
+        inferred Rhs_t: ComparableCollectionElement
```

Comment:
**Question** Do we need `ComparableCollectionElement` or would `EqualityCollectionElement` work instead (`CollectionElement` + `EqualityComparable`)?